87.md

```diff
@@ -0,0 +1,83 @@
+NIP-87
+=======
+
+Closed Communities
+------------------
+
+`draft` `optional` `author:staab` `author:earonesty`
+
+Some online communities may want a measure of privacy. While nostr's architecture isn't conducive to high levels of privacy due to metadata leakage and the difficulty of implementing ratchets for key rotation, it can provide basic encrypted channels appropriate for use with large groups with dynamic member lists.
+
+This is a different trade-off from NIP 24's encrypted chat, which is simpler and provides more privacy, but is inflexible and doesn't scale as well.
+
+This NIP relies on the following NIPs:
+
+- [86.md](NIP 86) for key sharing
+- [51.md](NIP 51) for member lists
+- [72.md](NIP 72) for community definition
+
+# Protocol Description
+
+## Community Definition
+
+Following NIP 72, communities are identified by the address of a replaceable `kind 34550` event. This may be wrapped using the shared key to support privately-defined communities.
+
+## Admin Key
+
+The key used to publish the community's `kind 34550` definition event MUST be used for performing privileged actions, such as publishing member lists and key rotations. Non-administrative events MAY be published to the community by the admin.
+
+Because the admin key is used for managing key rotation, it's recommended that a dedicated key be used for community administration instead of the community admin's personal key.
+
+## Key Sharing
+
+Key sharing and rotation is accomplished as described in [85.md](NIP 85).
+
+## Member Lists
+
+Admins MAY publish member lists using a kind `30000` event with the community's address as the `d` tag, as described in NIP 51. These events MAY be wrapped using NIP 59 to support private member lists.
+
```

Comment:
Are these Closed Communities for large groups? If so, consider an unbounded list (https://github.com/nostr-protocol/nips/pull/784) instead of a fixed list. Otherwise, this event can become very large and updating all the time.

Reply:
I strongly dislike nip 61. My preference would be for large groups to publish "users added" and "users removed" events instead. This same approach would fix the race conditions associated with lists (although not likely a problem in this case). But in the interest of a simple first draft I've used regular lists since they should still support several thousand members.